src/main/java/org/opensearch/ad/rest/RestValidateDetectorAction.java

```diff
@@ -1,44 +1,130 @@
+        this.sdkClient = extension.getClient();
+    }
     @Override
     public List<Route> routes() {
         return List.of(new Route(POST, "/detectors/_validate"), new Route(POST, "/detectors/_validate/{type}"));
     }
 
     @Override
-    public ExtensionRestResponse handleRequest(ExtensionRestRequest request) {
+    public ExtensionRestResponse handleRequest(ExtensionRestRequest request){
         if (!EnabledSetting.isADPluginEnabled()) {
             throw new IllegalStateException(CommonErrorMessages.DISABLED_ERR_MSG);
         }
-        Method method = request.method();
-
-        if (!Method.POST.equals(method)) {
-            return new ExtensionRestResponse(
-                request,
-                NOT_FOUND,
-                "Extension REST action improperly configured to handle " + request.toString()
-            );
+        AnomalyDetector detector;
+        XContentParser parser;
+        XContentBuilder builder = null;
+        ValidateAnomalyDetectorRequest validateAnomalyDetectorRequest;
+        try {
+            parser= request.contentParser(this.xContentRegistry);
+            ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+            String typesStr = validateTypeString(request);
+            DetectorValidationIssue issue=null;
+            try{
+                detector= AnomalyDetector.parse(parser);
+                //TODO Dependent on https://github.com/opensearch-project/opensearch-sdk-java/issues/211
+//                validateAnomalyDetectorRequest= new ValidateAnomalyDetectorRequest(
+//                        detector,
+//                        typesStr,
+//                        extension,
+//                        maxMultiEntityDetectors,
+//                        maxAnomalyFeatures,
+//                        requestTimeout
+//                );
+            }catch (Exception e){
+                if(e instanceof ADValidationException){
+                    ADValidationException ADException = (ADValidationException) e;
+                    issue = new DetectorValidationIssue(
+                            ADException.getAspect(),
+                            ADException.getType(),
+                            ADException.getMessage()
+                    );
+                }
+            }
+
+            try {
+                builder = XContentBuilder.builder(XContentType.JSON.xContent());
+                builder.startObject();
+                builder.field("issue", issue);
+                builder.endObject();
+            }catch (IOException e){
+                e.printStackTrace();
+            }
+
+
+
+        }catch (Exception e){
+            return new ExtensionRestResponse(request, BAD_REQUEST, builder);
         }
-        // do things with request
-        return new ExtensionRestResponse(request, OK, "placeholder");
+        return new ExtensionRestResponse(request, OK, builder);
```

Comment:
Looks like we are sending the same `builder` in the response for `BAD_REQUEST` and `OK` response.

Reply:
Actually once the logic will be implemented in validate detector, sending builder in response object will make more sense. But, yeah as of now we are keeping it as template I am replacing it with the "placeholder"